mars/tensor/core.py

```diff
@@ -780,5 +780,19 @@ def mutable_tensor(name, shape=None, dtype=np.float_, fill_value=None, chunk_siz
                                              fill_value=fill_value, chunk_size=chunk_size)
 
 
+def named_tensor(name, shape=None):
```

Comment:
Let `MarsDataset` support pass a name. Hide `named_tensor` for now until context can be used everywhere.